packages/hooks/use-post-type-infinite-scroll/index.tsx

```diff
@@ -0,0 +1,375 @@
+
+    // Aliases to needed state.
+    const current = state.source.get(link);
+    const firstLink = links[0];
+
+    const itemLinks = getLinksFromPages({ pages, firstLink, state });
+    const currentIndex = itemLinks.indexOf(link);
+    const nextLink = itemLinks[currentIndex + 1];
+
+    // Infinite scroll booleans.
+    const hasReachedLimit = !!limit && links.length >= limit;
+
+    const { supported, fetchRef, routeRef } = useInfiniteScroll({
+      currentLink: link,
+      nextLink,
+      fetchInViewOptions,
+      routeInViewOptions,
+    });
+
+    if (!current.isReady || isError(current)) return null;
+    if (!supported) return children;
+
+    const container = css`
+      position: relative;
+    `;
+
+    const fetcher = css`
+      position: absolute;
+      width: 100%;
+      bottom: 0;
+    `;
+
+    return (
+      <div css={container} ref={routeRef} className={className}>
+        {children}
+        {!hasReachedLimit && <div css={fetcher} ref={fetchRef} />}
+      </div>
+    );
+  };
+
+  return connect(Wrapper, { injectProps: false });
+};
+
+/**
+ * A memoized {@link Wrapper} to generate Wrapper components only once.
+ *
+ * @param link - The link for the post that the Wrapper belongs to.
+ * @param options - The intersection observer options for fetching and routing.
+ *
+ * @returns A React component that should be used to wrap the post.
+ */
+const MemoizedWrapper = memoize(
+  (link) => link,
+  (
+    link: string,
+    options: {
+      /**
+       * The intersection observer options for fetching.
+       */
+      fetchInViewOptions: IntersectionOptions;
+
+      /**
+       * The intersection observer options for routing.
+       */
+      routeInViewOptions: IntersectionOptions;
+    }
+  ) =>
+    wrapperGenerator({
+      link,
+      fetchInViewOptions: options.fetchInViewOptions,
+      routeInViewOptions: options.routeInViewOptions,
+    })
+);
+
+/**
+ * A hook used to add infinite scroll to any Frontity post type.
+ *
+ * @param options - Options for the hook, like the number of posts it should
+ * load autoamtically or if it's active or not. Defined in {@link
+ * UsePostTypeInfiniteScroll}.
+ *
+ * @returns - An array of posts and other useful booleans. Defined in {@link
+ * UsePostTypeInfiniteScroll}.
+ */
+const usePostTypeInfiniteScroll = (
+  options: UsePostTypeInfiniteScrollOptions = {}
+): UsePostTypeInfiniteScrollOutput => {
+  const defaultOptions = { active: true };
+  options = { ...defaultOptions, ...options };
+
+  const { state, actions } = useConnect<WpSource & Router>();
+  const { infiniteScroll } = state.router.state as InfiniteScrollRouterState;
+
+  // Values for browser state.
+  const archive: string = (() => {
+    // If `archive` is set in options, use it.
+    if (options.archive) {
+      return options.archive;
+    }
+
+    // If `archive` is already in the state, use it.
+    if (infiniteScroll?.archive) {
+      return infiniteScroll.archive;
+    }
```

Comment:
```suggestion
    // If `archive` is already in the state, use it.
    if (infiniteScroll?.archive) {
      return infiniteScroll.archive;
    }
    
    // If `archive` is set in options, use it.
    if (options.archive) {
      return options.archive;
    }
```

Shouldn't this be the other way around?

Reply:
I didn't change that part. If you think it makes more sense ―I think that too― I'll change it.

Maybe @orballo has something to say in this regard. :thinking:

Reply:
The question is: if `options.archive` is changed on the fly to a different archive, should the hook also change, or should the hook stick to the initial value of `options.archive`? 

And even before that question: if `options.archive` is changed on the fly to a different archive, will it break the hooks?

Reply:
Good question.

I think we should add a test for that case and see what happens. My guess is that it should not break, because `archive` is only used to initialize `pages` the first time the hook is executed (when the pages array is not defined yet). Therefore, the pages array is preserved and the hook should work the same.

https://github.com/frontity/frontity/blob/dec292d6d47a2a336e4e3c097e4c87d72a788e59/packages/hooks/use-post-type-infinite-scroll/index.tsx#L192-L195

My concern now is, what happen if you want to change the archive but `pages` already exists? I'll add a test for that as well.

Reply:
I've added two e2e tests more to check how the `usePostTypeInfiniteScroll()` hook behaves when the `archive` param changes. Both are passing without changing the code. https://github.com/frontity/frontity/pull/652/commits/60ddde54316f60285cfae65121ed34122a7986d0

Reply:
Awesome. Thank you David.